Add support for 'memstore', an in-memory session store

nerney opened the pull request:
While a cookie store keeps the session information in the cookies themselves, the other store options require use of a third-party service to store session data. 'memstore' is an in-memory store that stores session data in an in-memory map. 'memstore' is concurrency-safe. Perfect for testing/dev environments. Or, use-cases where simple cookie sessions are desired with session data server-side rather than in the cookies, and possible loss of sessions is not catastrophic.

Added a sub-packages 'memstore' and 'example-memstore'.
Added 'memstore' to readme and other documentation along with other stores.
Added 'memstore' to the vendors.

see - https://github.com/quasoft/memstore

nerney commented:
@appleboy @delphinus

delphinus commented on README.md:
```diff
@@ -28,7 +28,8 @@ import "github.com/gin-contrib/sessions"
 
-[embedmd]:# (example_cookie/main.go go)
+[embedmd]: # "example_cookie/main.go go"
+
```
Are these necessary diffs?  I think `[embedmd]: # (...)` is a special format for [embedmd](https://github.com/campoy/embedmd).

delphinus replied:
Ah, you did this commit for travis!  See [the log](https://travis-ci.org/gin-contrib/sessions/jobs/375171855), it says

<img width="163" alt="2018-05-09 8 57 53" src="https://user-images.githubusercontent.com/1239245/39788808-247e0dec-5367-11e8-9a43-251684a383d2.png">

This reports it discovered a redundant empty line here in README.  Fix it.

nerney replied:
fixed

delphinus commented on vendor/vendor.json:
```diff
@@ -1,169 +1,175 @@
-		}
-	],
-	"rootPath": "github.com/gin-contrib/sessions"
+  "comment": "",
```
Indents changed here (tab => 2 spaces?).  govendor did this?

appleboy replied:
please update this file via `govendor` command.

nerney replied:
i fixed all this

delphinus commented on .gitignore:
```diff
@@ -1,3 +1,3 @@
 coverage.out
 vendor/*
-!/vendor/vendor.json
+!/vendor/vendor.json
```
This diff seems also unnecessary.

appleboy replied:
revert this changes.

nerney replied:
done

nerney commented:
alright, @appleboy
i removed the whitespace change from .gitignore.
i reverted vendor.json and then used govendor to add the new dep, no more formatting stuff from my IDE.
i fixed the changes to readme that VSCode had made, embedmd syntax reverted and the new memstore section conforms with the existing syntax.
